### PR TITLE
Fix #15764: Implemented PMJ

### LINKED ISSUE #15764: pmj ##json
The JSON command for `pm` does not seem to be implemented?:

```
[0x00426870]> pm
0x00426870 1 Microsoft Cabinet archive data, 20011963 bytes, 7 files
[0x00426870]> pmj
failed r_magic_load ("j") no magic files loaded! could not find any magic files!
```

I would have expected an output such as:

```json
{"filetype": "Microsoft Cabinet archive data", "size": 20011963, "files": 7}
```

### PR DESCRIPTION
**Pull request by deepakchethan**
Screen Shot 2020-01-20 at 10 30 13 PM

### CODE REVIEW

File: libr/core/cmd_magic.c [hunk 1]
```diff
@@ -190,11 +190,14 @@ static int r_core_magic_at(RCore *core, const char *file, ut64 addr, int depth,
...
 
-static void r_core_magic(RCore *core, const char *file, int v) {
+static void r_core_magic(RCore *core, const char *file, int v, int json) {
 	ut64 addr = core->offset;
 	int hits = 0;
 	magicdepth = r_config_get_i (core->config, "magic.depth"); // TODO: do not use global var here
-	r_core_magic_at (core, file, addr, magicdepth, v, false, &hits);
+	r_core_magic_at (core, file, addr, magicdepth, v, json, &hits);
+	if (json) {
+		r_cons_newline ();
```
Comment (Maijin) [inline libr/core/cmd_magic.c]: You sure you pushed everything?

Comment (deepakchethan) [reply libr/core/cmd_magic.c]: "r_core_magic_at" already supported json output. I just had to add a newline at the end to make sure that the printed json is not cleared

**Comment by deepakchethan** [comment]
This implementation is based on https://github.com/radareorg/radare2/issues/15764#issuecomment-570915563

File: libr/core/cmd_magic.c (hunk 1 above)
Comment (Maijin) [reply libr/core/cmd_magic.c]: ah nice :) Can you add test in json tests?

Comment (deepakchethan) [reply libr/core/cmd_magic.c]: Sure, will add the test

**Pull request merged**